src/lsptoolshost/roslynLanguageServer.ts

```diff
@@ -202,6 +205,46 @@ export class RoslynLanguageServer {
                 workspace: {
                     configuration: (params) => readConfigurations(params),
                 },
+                resolveCodeAction: async (codeAction, token, next) => {
+                    const lspCodeAction = <CodeAction>codeAction;
+                    const data = lspCodeAction.data;
+                    if (data.FixAllFlavors) {
+                        const result = await vscode.window.showQuickPick(data.FixAllFlavors, {
+                            placeHolder: 'Pick a fix all scope',
+                        });
+
+                        if (result) {
+                            const fixAllCodeAction: RoslynProtocol.RoslynFixAllCodeAction = {
+                                title: lspCodeAction.title,
+                                edit: lspCodeAction.edit,
+                                data: data,
+                                scope: result,
+                            };
+
+                            const response = await _languageServer.sendRequest(
```

Comment:
it could likely take some time to compute this, especially for a solution level fix all in a large solution.  We definitely should wrap this entire operation in a cancellable progress.  See this https://github.com/dotnet/vscode-csharp/blob/main/src/lsptoolshost/unitTesting.ts#L84 for an example on how to do it
Some notes
1.  We don't have an explicit progress % for this, so fine to report progress without an increment.
2.  We should allow cancellation up until we call `vscode.workspace.applyEdit` - that part should not be cancellable.